### DISCUSSION TITLE
Unable to commit on doom emacs

### DISCUSSION THREAD

**Discussion by Vanderscycle** [magit/magit#4893] [Support] [answered]
Hey I need some help as I am at my wits ends with this weird bug/help request.
I posted on r/emacs, doomemacs discord server, and other emacs users that I know. I have spent countless of hours searching reddit, so, magit issues and couldn't find a way to commit.  I removed magit from doom, purged, synced and reinstalled magit. I can however commit on the cli using `lazygit` 

I am still new to emacs and the more I learn about emacs/elisp the more I enjoy it. 

(1) Able to commit using magit using a gpg key

(2) Unable to commit (I assume) but can push and pull with my ssh

I can still write the commit message but pressing `:x` creates this error when trying with my gpg key
```
  1 git --no-pager --literal-pathspecs -c core.preloadindex\=true -c log.showSignature\=false -c color.ui\=false -c color.diff\=false commit --gpg-sign\=EE8056E0B943FBDE374817902664645CF72BE38A
hint: Waiting for your editor to close the file...
Waiting for Emacs...
*ERROR*: Wrong type argument: keymapp, nil
```

Without requiring gpg signature
```
  1 git --no-pager --literal-pathspecs -c core.preloadindex\=true -c log.showSignature\=false -c color.ui\=false -c color.diff\=false commit --
hint: Waiting for your editor to close the file...
Waiting for Emacs...
*ERROR*: Wrong type argument: keymapp, nil
```
(3) My setup
My gpg-agent.conf
```
# Do not ask me about entered passwords for 24h (during the same session)
default-cache-ttl 86400
max-cache-ttl 86400

pinentry-program /usr/bin/pinentry-tty
allow-loopback-entry
# Allow pinentry in Emacs minibuffer (combined with epg-pinentry-mode)
allow-loopback-pinentry
allow-emacs-pinentry
```

The config.el
```elisp
(after! magit
  ;; makes magit prompt me for the new branch name first:
  (setq magit-branch-read-upstream-first nil))

(after! epg
  (setq epg-pinentry-mode nil))

```
package.el
```elisp
(package! pinentry)
```

Thank you for any help

**Comment by tarsius** [comment, marked as answer]
The fact that you are signing your commit *might* be unrelated. Try creating a commit without signing it, just in case.

Since the error occurs when you press `C-c C-c` to finish the commit, not when pressing `c c` to initiate it; you should be able to get a backtrace. That should help figuring out which keymap borked. `M-x toggle-debug-on-error`, then try again.

**Comment by Vanderscycle** [comment]
Hey thanks for getting back,

I disabled the `gpgsign` in my `.gitconfig` and the `toggle-debug-on-error` isn't yielding an expanded message when using `C-c C-c` or `c c` just the aforementioned messages. However, I decided to copy the git command used by emacs `git --no-pager --literal-pathspecs -c core.preloadindex\=true -c log.showSignature\=false -c color.ui\=false -c color.diff\=false commit --` which yielded this message.

```elisp
Debugger entered--Lisp error: (error "Unknown option ‘--wait’")
  signal(error ("Unknown option ‘--wait’"))
  error("Unknown option `%s'" "--wait")
  #f(compiled-function (args-left) "A subroutine of `command-line'." #<bytecode -0x37c05ab03352e2d>)(("--wait" "/home/henri/Documents/infrastructure/.git/COMMIT_E..."))
  apply(#f(compiled-function (args-left) "A subroutine of `command-line'." #<bytecode -0x37c05ab03352e2d>) ("--wait" "/home/henri/Documents/infrastructure/.git/COMMIT_E..."))
  command-line-1(("--wait" "/home/henri/Documents/infrastructure/.git/COMMIT_E..."))
  command-line()
  normal-top-level()

```

**Reply by tarsius** [reply]
It seems you are using `emacsclient --wait ...` but that command does not take a `--wait` option. (See `emacsclient --help`.)

**Reply by Vanderscycle** [reply]
I am a tad confused as I never gave `emacsclient` the `--wait` command. Do you know where else I could investigate on emacslient?

**Comment by Vanderscycle** [comment]
I purged doom emacs and emacs completly and started over at 0 and it now works.

@tarsius thank you for your willingness to help me in my madness.

Take care,

**Reply by tarsius** [reply]
You're welcome!
